### DISCUSSION TITLE
Thanks for the amazing library

### DISCUSSION THREAD

**Discussion by SuryaViswanath** [pymupdf/PyMuPDF#754] [General]
I want to thank you for creating this amazing module for pdf's 

I just wanted to start this thread so that if anyone wants to share their view's about this amazing library it would be nice.

**Comment by victor-ab** [comment]
Thanks not just for the library itself, but also for being very responsive (and having a lot of patience) at any opened issue.

@JorjMcKie, I also saw your work with Nuitka, you are amazing. Hope I can create amazing things and live my retirement in some caribbean island as well.

**Reply by JorjMcKie** [reply]
My best wishes are with you - I'm sure you will achieve your goals.

**Comment by zufj** [comment]
@JorjMcKie I started using the PyMuPDF module because of its speed - and kept using it because of its features and regular improvements :+1:  Thank you very much for your efforts and dedication!

**Comment by ecko4life** [comment]
Firstly, I would like to thank @JorjMcKie for being so helpful in providing explanations and the way of such a library function.

I had some open questions in regards to how to get the sequence of images from the file (when exporting them). 

Due to the fact that many pdf formats seem to be a different structure, it is highly recommended to use cleanContents() method. After that, it's important to locate each img thanks to getImageBbox()! 
 
Thanks to the library we're able to get the desired outputs! Much appreciation.
